examples/raytracing-rchit/shader.rchit

```diff
@@ -0,0 +1,11 @@
+#version 460 core
+#extension GL_EXT_ray_tracing : require
+
+layout(location = 0) rayPayloadInEXT vec3 hitPayload;
+
+hitAttributeEXT vec3 normal; // read-only
```

Comment:
Are `hitAttributeEXT`s always read-only, or just this particular one? If they are always read-only, then they shouldn't be included as part of the out interface. If this specific instance is only read, then I think that is obvious from how small the shader is. In either case, I think you should remove the comment.

Reply:
I found this, about `HitAttributeNV`:

> They can be written to only in IntersectionNV execution model and read from only in AnyHitNV and ClosestHitNV execution models.

I recommend adding another parameter to `ioGen` which saves the shader type. This should help us determine whether the variable is readable/writeable.